corehq/apps/reports/standard/sms.py

```diff
@@ -1103,4 +1109,177 @@ class PhoneNumberReport(BaseCommConnectLogReport):
+
+    @property
+    @memoized
+    def phone_number_filter(self):
+        value = self._filter['phone_number_filter']
+        if isinstance(value, basestring):
+            return apply_leniency(value.strip())
+
+        return None
+
+    @property
+    def contact_type(self):
+        return self._filter['contact_type']
+
+    @property
+    def selected_group(self):
+        return self._filter['selected_group']
+
+    @property
+    @memoized
+    def user_ids_in_selected_group(self):
+        return Group.get(self.selected_group).users
+
+    @property
+    def has_phone_number(self):
+        return self._filter['has_phone_number']
+
+    @property
+    def verification_status(self):
+        return self._filter['verification_status']
+
+    @property
+    def _show_users_without_phone_numbers(self):
+        return (
+            self.filter_type == 'contact' and
+            self.contact_type == 'users' and
+            self.has_phone_number != 'has_phone_number'
+        )
+
+    @property
+    def _show_cases(self):
+        return self.contact_type == 'cases'
+
+    def _fmt_owner(self, owner_doc_type, owner_id, owner_cache):
+        doc_info = self.get_recipient_info(owner_doc_type, owner_id, owner_cache)
+        table_cell = self._fmt_contact_link(owner_id, doc_info)['html']
+        return table_cell
+
+    def _fmt_status(self, number):
+        if number.verified:
+            return "Verified"
+        elif number.pending_verification:
+            return "Verification Pending"
+        elif not (number.is_two_way or number.pending_verification) and PhoneNumber.get_reserved_number(number):
+            return "Already in use"
+        return "Not Verified"
+
+    def _fmt_row(self, number, owner_cache):
+        if isinstance(number, PhoneNumber):
+            return [
+                self._fmt_owner(number.owner_doc_type, number.owner_id, owner_cache),
+                number.phone_number,
+                self._fmt_status(number),
+                "Yes" if number.is_two_way else "No",
+            ]
+
+        return [
+            self._fmt_owner(number.owner_doc_type, number.owner_id, owner_cache),
+            '---',
+            '---',
+            '---',
+        ]
+
+    def _get_rows(self, paginate=True):
+        owner_cache = {}
+        if self._show_users_without_phone_numbers:
+            data = self._get_users_without_phone_numbers()
+        else:
+            data = self._get_queryset()
+
+        if paginate and self.pagination:
+            data = data[self.pagination.start:self.pagination.start + self.pagination.count]
+
+        for number in data:
+            yield self._fmt_row(number, owner_cache)
+
+    def _get_queryset(self):
+        query = PhoneNumber.objects.filter(domain=self.domain)
+
+        if self.filter_type == 'phone_number':
+            if self.phone_number_filter:
+                query = query.filter(phone_number__contains=self.phone_number_filter)
```

Comment:
yeah i think you're right. we're going to have to do some cleanup with that report anyway, will add that to the list